test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -156,10 +159,12 @@ class LogicalDnsClusterTest : public testing::Test {
     expectResolve(Network::DnsLookupFamily::V4Only, expected_address);
     resolve_timer_->invokeCallback();
 
-    // Empty should not cause any change.
+    // Failure should not cause any change.
+    // TODO(junr03): make sure to have coverage here. Why does strict change but logical not? Why
+    // this inconsistency? Updated in subsequent change.
```

Comment:
See my other comment. IMO logical DNS should "stabilize" but strict should not in the case of a successful nothing response, but let's make sure to clearly document this.

Reply:
chatted about this offline, as I was thinking that the suggested behavior change was a little odd. I am now clear on the expectation.

We also agreed that at this point we can just do #9927 here. I will update this today.